test/integration/middleware_test.rb

```diff
@@ -22,4 +22,28 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
       Rails.configuration.webpack.dev_server.enabled = false
     end
   end
+
+  test "webpack dev server adds the dev server to Content-Security-Policy on pages triggering a welcome page" do
+    begin
+      Rails.configuration.webpack.dev_server.enabled = true
+      Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+      Environment.stubs(:first).returns(nil)
```

Comment:
Both new tests are missing any verification that these stubs have triggered the filters you expect. Probably at a minimum they should be expectations, but some kind of verification of the page content may also help.